follow_fork/follow_fork.c

```diff
@@ -0,0 +1,124 @@
+/**
+ * Allows the LiveRecorder API to follow fork() and record
+ * all children processes.
+ *
+ * See README.md for how to compile and use.
+ *
+ */
+#define _POSIX_SOURCE
+#define _GNU_SOURCE
+
+#include "undolr.h"
+#include "undolr_deprecated.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdbool.h>
+#include <time.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <linux/limits.h>
+
+extern char *__progname;
+
+void attach_lr(int, char**, char**) __attribute__ ((constructor));
+
+static bool
+process_needs_recording(pid_t pid)
+{
+    if (strncmp(__progname, "undo", 4) == 0)
+    {
+        if (unsetenv("LD_PRELOAD") == -1)
```

Comment:
I think that doing `if (unsetenv(...))` is better even if `unsetenv` should only return 0 or -1.

Reply:
Agreed, slightly more readable